### PR TITLE
[Example]: Dflash-Offline Launcher Example

### PR DESCRIPTION
**Pull request by h-guo18**
### What does this PR do?

Type of change: new example

Offline DFlash training launcher example for Qwen3-0.6B. Two-task pipeline: dump base-model hidden states via HF forward, then train DFlash on the dump.

- `tools/launcher/examples/Qwen/Qwen3-0.6B/hf_offline_dflash.yaml` — new launcher YAML
- `tools/launcher/common/eagle3/dump_offline_data_hf.sh` — new HF-backed dump script (DFlash's `answer_only_loss=true` requires `loss_mask`, which the existing TRT-LLM dump backend does not produce)
- `examples/dataset/synthetic_conversations_1k.jsonl` — `conversation_id` added at-source (asserted by `compute_hidden_states_*.py`; previously injected at test-fixture time)
- `tests/regression/torch/speculative/test_dflash_offline.py` — `tagged_synth_data_path` fixture removed since the field is now at-source

### Usage

```
uv run launch.py --yaml examples/Qwen/Qwen3-0.6B/hf_offline_dflash.yaml --yes
```

### Testing

End-to-end on CoreWeave Slurm (1×1 GPU): task_0 dump succeeded; task_1 training loss 7.85 → 2.94 over 2 epochs, regression PASSED.

### Before your PR is "*Ready for review*"

- Is this change backward compatible?: ✅ (`conversation_id` is purely additive on the dataset)
- If you copied code from any other sources or added a new PIP dependency, did you follow guidance in `CONTRIBUTING.md`: N/A
- Did you write any new necessary tests?: N/A (existing `test_dflash_offline.py` covers the workflow)
- Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: N/A (new example only)
- Did you get Claude approval on this PR?: ❌

## Summary by CodeRabbit

* **New Features**
  * Added launcher script for offline hidden-state processing with HuggingFace backend support and distributed execution capabilities
  * Added training configuration example for Qwen3-0.6B model with DFlash speculative decoding pipeline

* **Tests**
  * Updated offline hidden-states regression test to improve data handling workflow

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/Model-Optimizer/pull/1529?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Review by shengliangxu — APPROVED** [review]
LGTM
